jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/SpringBHVisitorLayoutAlgorithm.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2003, The JUNG Authors
+ * All rights reserved.
+ *
+ * This software is open-source under the BSD license; see either "license.txt"
+ * or https://github.com/jrtom/jung/blob/master/LICENSE for a description.
+ */
+package edu.uci.ics.jung.layout.algorithms;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.graph.EndpointPair;
+import com.google.common.graph.Graph;
+import edu.uci.ics.jung.algorithms.util.IterativeContext;
+import edu.uci.ics.jung.layout.model.LayoutModel;
+import edu.uci.ics.jung.layout.model.Point;
+import edu.uci.ics.jung.layout.spatial.BarnesHutQuadTree;
+import edu.uci.ics.jung.layout.spatial.ForceObject;
+import java.util.ConcurrentModificationException;
+import java.util.Optional;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The SpringLayout package represents a visualization of a set of nodes. The SpringLayout, which is
+ * initialized with a Graph, assigns X/Y locations to each node. When called <code>relax()</code>,
```

Comment:
I think you mean "step()" :)

Reply:
who, me? :)
This javadoc is unchanged since version 2.0 in 2009. I don't have the version 1 code handy to look farther back... You might need to ask Danyel about it!

Reply:
Ah, hadn't noticed it was a direct copy.  Never mind, we'll hopefully catch that in the next documentation pass.